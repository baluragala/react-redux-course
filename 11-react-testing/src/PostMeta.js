import React, {Component} from 'react';


class PostMeta extends Component {

  render() {
    const {author,createdAt} = this.props;
    return (
      <div>
        <div className="post-meta">
          <a>
            <img src={author.image} role="presentation"/>
          </a>
          <div className="info">
            <a className="author">{author.username}</a>
            <span className="date">{createdAt}</span>
          </div>
          {/*<favorite-button className="pull-xs-right">
            <button className="btn btn-sm btn-outline-primary">
              <i className="ion-heart"/>0
            </button>
          </favorite-button>*/}
        </div>
      </div>
    )
  }
}

export default PostMeta;


/*
 <div class="article-meta">
 <a ng-reflect-router-link="/profile,lloyd" ng-reflect-href="#/profile/lloyd" href="#/profile/lloyd">
 <img ng-reflect-src="https://pbs.twimg.com/profile_images/722819453939888130/osWwzkFY_bigger.jpg" src="https://pbs.twimg.com/profile_images/722819453939888130/osWwzkFY_bigger.jpg">
 </a>

 <div class="info">
 <a class="author" ng-reflect-router-link="/profile,lloyd" ng-reflect-href="#/profile/lloyd" href="#/profile/lloyd">
 lloyd
 </a>
 <span class="date">
 March 4, 2017
 </span>
 </div>


 <favorite-button class="pull-xs-right" ng-reflect-article="[object Object]"><button class="btn btn-sm btn-outline-primary" ng-reflect-klass="btn btn-sm" ng-reflect-ng-class="[object Object]">
 <i class="ion-heart"></i>
 0

 </button>
 </favorite-button>

 </div>*/
