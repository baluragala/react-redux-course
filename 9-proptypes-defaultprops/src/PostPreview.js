import React, {Component} from 'react';
import {Link} from 'react-router';
import './PostItem.css';

import PostMeta from './PostMeta';

class PostPreview extends Component {

  static propTypes = {
    post : React.PropTypes.shape({
      id: React.PropTypes.number.isRequired,
      title: React.PropTypes.string.isRequired,
      author: React.PropTypes.object.isRequired,
    }).isRequired
  }

  static defaultProps = {
    post : {
      id:0,
      title:'Title comes here',
      author:{}
    }
  };


  render() {
    const {post} = this.props;
    console.log(React.PropTypes);
    return (
      <div className="panel panel-default already-visible come-in">
        <div className="panel-heading"><Link to={"posts/" + post.id }>{post.title}</Link></div>
        <div className="panel-body">
          <PostMeta author={post.author} createdAt={post.createdAt}/>
          <p style={{margin: '5px'}}>{post.description}</p>
        </div>
      </div>
    );
  }
}

export default PostPreview;
