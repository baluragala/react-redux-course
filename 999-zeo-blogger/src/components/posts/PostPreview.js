import React, {Component} from 'react';
import {Link} from 'react-router';
import './PostItem.css';

import PostMeta from './PostMeta';

class PostPreview extends Component {

  render() {
    const {post} = this.props;
    console.log(post);
    return (
      <div className="panel panel-default already-visible come-in">
        <div className="panel-heading"><Link to={"/posts/" + post.id }>{post.title}</Link></div>
        <div className="panel-body">
          <PostMeta author={post.author} createdAt={post.createdAt}/>
          <p style={{margin: '5px'}}>{post.description}</p>
        </div>
      </div>
    );
  }
}

export default PostPreview;
