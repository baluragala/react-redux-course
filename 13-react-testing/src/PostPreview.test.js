import React from 'react';
import {shallow} from 'enzyme';
import PostPreview from './PostPreview';
import PostMeta from './PostMeta'
import {Link} from 'react-router';
import renderer from 'react-test-renderer';

describe('<PostPreview />', () => {

  it('should render an PostMeta', () => {
    const wrapper = shallow(<PostPreview
      post={{id: 1, title: 'test', body: 'test', description: 'test', author: {image: '', bio: ''}}}/>);
    //console.log(wrapper.debug());
    expect(wrapper.find(PostMeta)).toBeDefined();
  });

  it('should render Link to PostDetail with id 1', () => {
    const wrapper = shallow(<PostPreview
      post={{id: 1, title: 'test', body: 'test', description: 'test', author: {image: '', bio: ''}}}/>);
    //console.log(wrapper.debug());
    let link = wrapper.find(Link);
    expect(link.prop('to')).toEqual('posts/1');
  });


  it('renders correctly', () => {
    const tree = renderer.create(
      <PostPreview
        post={{id: 1, title: 'test', body: 'test', description: 'test', author: {image: '', bio: ''}}}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});