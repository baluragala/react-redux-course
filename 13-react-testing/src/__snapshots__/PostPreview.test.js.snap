exports[`<PostPreview /> renders correctly 1`] = `
<div
  className="panel panel-default already-visible come-in">
  <div
    className="panel-heading">
    <a
      onClick={[Function]}
      style={Object {}}>
      test
    </a>
  </div>
  <div
    className="panel-body">
    <div>
      <div
        className="post-meta">
        <a>
          <img
            role="presentation"
            src="" />
        </a>
        <div
          className="info">
          <a
            className="author" />
          <span
            className="date" />
        </div>
      </div>
    </div>
    <p
      style={
        Object {
          "margin": "5px",
        }
      }>
      test
    </p>
  </div>
</div>
`;
